NotesApplication/Screens/Home/NotesViewModel.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  NotesViewModel.swift
+//  NoteApp
+//
+//  Created by Ana Ioramashvili on 31.10.24.
+//
+
+import Foundation
+
+protocol NotesViewModelDelegate: AnyObject {
+    func notesDidUpdate()
+}
+
+final class NotesViewModel {
+    // MARK: - Properties
+    private(set) var notes: [CoreDataNote] = []
+    weak var delegate: NotesViewModelDelegate?
+    
+    // MARK: - Init
+    init() {
+        fetchNotes()
+    }
+    
+    // MARK: - Public Methods
+    func fetchNotes() {
+        notes = DataComunication.shared.fetchNotes()
+        delegate?.notesDidUpdate()
+    }
+    
+    func addNote(title: String, content: String) {
+        DataComunication.shared.addNote(title: title, content: content)
+        fetchNotes()
+    }
+    
+    func deleteNote(at index: Int) {
+        let note = notes[index]
+        DataComunication.shared.deleteNote(note)
+        fetchNotes()
+    }
+    
+    func updateNote(at index: Int, title: String, content: String) {
+        DataComunication.shared.updateNote(at: index, title: title, content: content)
+        fetchNotes()
+    }
+    
+    func numberOfNotes() -> Int {
```

Comment:
insead of this you can have 
var numberOfNotes: Int {
        notes.count
    } and in tableview you can use as viewmodel. numberOfNotes but it is up to you